identicon.js

```diff
@@ -53,11 +55,16 @@
                 image   = new PNGlib(size, size, 256);
 
             // light-grey background
-            var bg      = image.color(this.background[0], this.background[1], this.background[2], this.background[3]);
-
-            // foreground is last 7 chars as hue at 50% saturation, 70% brightness
-            var rgb     = this.hsl2rgb(parseInt(hash.substr(-7), 16) / 0xfffffff, 0.5, 0.7),
-                fg      = image.color(rgb[0] * 255, rgb[1] * 255, rgb[2] * 255);
+            var bg      = image.color(this.background[0], this.background[1], this.background[2], this.background[3]),
+                fg;
+
+            if(this.foreground) {
```

Comment:
More nitpicking here. Can we add a space between if and the opening parenthesis to match style? Also if you could line up the assignments that would be lovely.